system-addon/bootstrap.js

```diff
@@ -0,0 +1,91 @@
+/* globals Components, XPCOMUtils, Preferences, ActivityStream */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Preferences.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "ActivityStream",
+  "resource://activity-stream/lib/ActivityStream.jsm");
+
+const ACTIVITY_STREAM_ENABLED_PREF = "browser.newtabpage.activity-stream.enabled";
+const ACTIVITY_STREAM_OPTIONS = {newTabURL: "resource://activity-stream/data/activity-stream.html"};
+const REASON_STARTUP_ON_PREF_CHANGE = "PREF_ON";
+const REASON_SHUTDOWN_ON_PREF_CHANGE = "PREF_OFF";
+
+let activityStream;
+let startupData;
+
+/**
+ * init - Initializes an instance of ActivityStream. This could be called by
+ *        the startup() function exposed by bootstrap.js, or it could be called
+ *        when ACTIVITY_STREAM_ENABLED_PREF is changed from false to true.
+ *
+ * @param  {string} reason - Reason for initialization. Could be install, upgrade, or pref change
+ */
+function init(reason) {
+  // Don't re-inlialize
+  if (activityStream && activityStream.initialized) {
+    return;
+  }
+  const options = Object.assign({}, startupData || {}, ACTIVITY_STREAM_OPTIONS);
+  activityStream = new ActivityStream(options);
+  activityStream.init(reason);
+}
+
+/**
+ * uninit - Uninitializes the activityStream instance, if it exsits.This could be
+ *          called by the shutdown() function exposed by bootstrap.js, or it could
+ *          be called when ACTIVITY_STREAM_ENABLED_PREF is changed from true to false.
+ *
+ * @param  {type} reason Reason for uninitialization. Could be uninstall, upgrade, or pref change
+ * @return {type}        description
+ */
+function uninit(reason) {
+  if (activityStream) {
+    activityStream.uninit(reason);
+    activityStream = null;
+  }
+}
+
+/**
+ * onPrefChanged - handler for changes to ACTIVITY_STREAM_ENABLED_PREF
+ *
+ * @param  {bool} isEnabled Determines whether
+ */
+function onPrefChanged(isEnabled) {
+  if (isEnabled) {
+    init(REASON_STARTUP_ON_PREF_CHANGE);
+  } else {
+    uninit(REASON_SHUTDOWN_ON_PREF_CHANGE);
+  }
+}
+
+// The functions below are required by boostrap.js
+
+this.install = function install() {};
```

Comment:
This wants params, I think, just to keep us from later tripping on the strange behavior documented on https://developer.mozilla.org/en/Add-ons/Bootstrapped_extensions#uninstall